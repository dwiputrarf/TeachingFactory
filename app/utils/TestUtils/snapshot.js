/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';

const mockSetState = Component =>
  class extends Component {
    setState(f, callback) {
      if (typeof f === 'function') {
        /* eslint-disable react/no-direct-mutation-state */
        this.state = { ...this.state, ...f(this.state) };
      } else {
        /* eslint-disable react/no-direct-mutation-state */
        this.state = { ...this.state, ...f };
      }
      if (callback) callback();
    }
  };

const assertSnapshot = (C, config) => {
  const { props, state, desc, throws, beforeTest } = config;

  let Component;
  let RenderedComponent;

  if (beforeTest) {
    beforeTest();
  }

  if (throws) {
    test(desc, () => {
      expect(() => {
        if (C.prototype.render) {
          Component = new C(props);
        } else {
          Component = C(props);
        }
      }).toThrow(throws);
    });
  } else {
    if (C.prototype.render) {
      const CWithMockState = mockSetState(C);
      Component = new CWithMockState(props);

      if (state) {
        Component.setState(state);
      }

      RenderedComponent = Component.render();
    } else {
      RenderedComponent = C(props);
    }

    test(desc, () => {
      const tree = RenderedComponent && renderer.create(RenderedComponent).toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
};

const assertSnapshots = (Component, configs) => {
  configs.forEach(config => assertSnapshot(Component, config));
};

export { assertSnapshots, mockSetState };
export default { assertSnapshots, mockSetState };
