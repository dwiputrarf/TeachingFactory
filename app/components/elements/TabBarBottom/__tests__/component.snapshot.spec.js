import { assertSnapshots } from '../../../../utils/TestUtils/snapshot';
import Component from '../index';

jest.mock('react-navigation', () => ({
  TabBarBottom: 'TabBarBottom'
}));

describe('Component snapshot testing', () => {
  const configs = [
    {
      state: { isVisible: false },
      props: {},
      desc: 'renders with default props'
    },
    {
      state: { isVisible: true },
      props: {},
      desc: 'renders with isVisible = true'
    }
  ];
  assertSnapshots(Component, configs);
});
