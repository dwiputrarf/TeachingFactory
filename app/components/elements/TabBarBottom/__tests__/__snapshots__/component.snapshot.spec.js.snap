// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot testing renders with default props 1`] = `null`;

exports[`Component snapshot testing renders with isVisible = true 1`] = `<TabBarBottom />`;
